packages/dashboard/src/vaadin-dashboard-layout-mixin.js

```diff
@@ -39,6 +41,30 @@ export const DashboardLayoutMixin = (superClass) =>
             minmax(var(--_vaadin-dashboard-col-min-width), var(--_vaadin-dashboard-col-max-width))
           );
         }
+
+        ::slotted(*) {
+          grid-column: span min(var(--vaadin-dashboard-item-colspan, 1), var(--_vaadin-dashboard-item-max-colspan));
+        }
       `;
     }
+
+    /**
+     * @protected
+     * @override
+     */
+    _onResize() {
+      this.__updateItemMaxColspan();
+    }
+
+    /**
+     * @private
+     */
+    __updateItemMaxColspan() {
+      // Temporarily set max colspan to 1
+      this.style.setProperty('--_vaadin-dashboard-item-max-colspan', 1);
```

Comment:
This seems to cause flickering when resizing.

Reply:
https://github.com/user-attachments/assets/837ac813-76c3-43ac-a418-573754a8f9de

Reply:
Sure, a `ResizeObserver` isn't synchronous (and there is no CSS-only way to implement the colspan capping). We can later try to improve this by, for example, wrapping the content inside another container that we manually resize to match the host's size.

Reply:
I added an [issue](https://github.com/orgs/vaadin/projects/70/views/1?pane=issue&itemId=75035546) for this to the ideas column.

Reply:
> wrapping the content inside another container that we manually resize to match the host's size

Did a quick prototype of this here https://github.com/vaadin/web-components/tree/proto/dashboard-container and it does eliminate the flickering